client/v2/client.go

```diff
@@ -565,24 +565,29 @@ func (c *client) Query(q Query) (*Response, error) {
 	}
 	defer resp.Body.Close()
 
+	// check status code before doing any JSON marshalling.
+	// if the status code is >= 400, there won't be any valid json to marshal so
+	// return immediately.
+	if resp.StatusCode >= 400 {
+		return nil, fmt.Errorf("Received bad status code from server: %d (%s)",
+			resp.StatusCode, resp.Status)
```

Comment:
For consistency, can you change the initial letter to lowercase? Also on the other `Errorf()` calls below.